src/main/scala/org/apache/spark/sql/delta/metering/DeltaLogging.scala

```diff
@@ -102,4 +103,23 @@ trait DeltaLogging
       new OpType(opType, ""),
       extraTags = tableTags ++ tags) {thunk}
   }
+  /**
+   * Wrap various delta operations to provide a more meaningful name in Spark UI
+   * @param desc a short description of the operation
+   */
+  def withJobDescription[U](desc: String, session: SparkSession)(body: => U): U = {
+    val sc = session.sparkContext
+    val oldDesc = sc.getLocalProperty(SparkContext.SPARK_JOB_DESCRIPTION)
+    val suffix = if (oldDesc == null) {
+      ""
+    } else {
+      s" $oldDesc:"
+    }
+    try {
+      sc.setJobDescription(s"Delta:$suffix $desc")
```

Comment:
In a nested invocation, this may seem a bit weird. Do we need to include the parent invocation's description?

Reply:
Thanks for the feedback @brkyvz 
In an app where many Delta based queries are submitted, this helps correlate the "housekeeping" queries with the top level user query (in Spark 3.0 jobDescription is the SQL text).
In Delta code I made the change such that `withJobDescription` calls are not nested.  So I think the `suffix` is useful.

The other comments make sense, I'll make the adjustments.

Reply:
ah got it. It's not nested with respect to Delta but something else.